include/wil/cppwinrt_authoring.h

```diff
@@ -330,5 +330,117 @@ struct single_threaded_notifying_property : single_threaded_rw_property<T>
  */
 #define INIT_NOTIFYING_PROPERTY(NAME, VALUE) NAME(&m_propertyChanged, *this, L"" #NAME, VALUE)
 
+namespace details
+{
+#ifdef WINRT_Microsoft_UI_Xaml_Data_H
+    using Xaml_DependencyProperty = winrt::Microsoft::UI::Xaml::DependencyProperty;
+    using Xaml_PropertyChangedCallback = winrt::Microsoft::UI::Xaml::PropertyChangedCallback;
+    using Xaml_PropertyMetadata = winrt::Microsoft::UI::Xaml::PropertyMetadata;
+#elif defined(WINRT_Windows_UI_Xaml_Data_H)
+    using Xaml_DependencyProperty = winrt::Windows::UI::Xaml::DependencyProperty;
+    using Xaml_PropertyChangedCallback = winrt::Windows::UI::Xaml::PropertyChangedCallback;
+    using Xaml_PropertyMetadata = winrt::Windows::UI::Xaml::PropertyMetadata;
+#endif
+
+    using Xaml_DependencyObject_GetValue = std::function<winrt::Windows::Foundation::IInspectable(Xaml_DependencyProperty const&)>;
+    using Xaml_DependencyObject_SetValue = std::function<void(Xaml_DependencyProperty const&, winrt::Windows::Foundation::IInspectable const&)>;
+
+    template <typename PropertyType, typename OwnerType>
+    inline Xaml_DependencyProperty register_dependency_property(
+        const std::wstring_view& propertyNameString)
+    {
+        return Xaml_DependencyProperty::Register(
+            propertyNameString,
+            winrt::template xaml_typename<PropertyType>(),
+            winrt::template xaml_typename<OwnerType>(),
+            nullptr);
+    }
+
+    template <typename PropertyType, typename OwnerType, typename DefaultValueType = PropertyType>
+    inline Xaml_DependencyProperty register_dependency_property(
+        const std::wstring_view& propertyNameString,
+        const DefaultValueType& defaultValue,
+        const Xaml_PropertyChangedCallback& propertyChangedCallback = nullptr)
+    {
+        // TODO: assert T and PropertyType are compatible

```

Comment:
put a `typename = std::enable_if_t<is_assignable_v<PropertyType, DefaultValueType>>` to filter it out

Reply:
Suggestion: require C++20 and use `std::convertible_to<PropertyType> DefaultValueType`

Nit: C++/WinRT has no formal documented way to 100% check this, see https://github.com/microsoft/cppwinrt/issues/609